_posts/articles/2022-07-11-sysdig-adopting-docs-as-code.md

```diff
@@ -0,0 +1,104 @@
+
+## Hackathon Effect
+
+A Hackathon is a time-boxed event in which cross-functional teams work on creative ideas. Sysdig conducts a yearly Hackathon fest where employees can propose innovative ideas and collaborate with colleagues who they don’t otherwise get to work closely with.
+
+The impossible becomes possible with the Sysdig Hackathon. Our team proposed the idea of porting the XML content in Paligo to the Doc-as-Code scheme using an innovative in-house tool.
+
+As [Parkinson’s law](https://en.wikipedia.org/wiki/Parkinson's_law) states, “work expands so as to fill the time available for its completion.” Thanks to the Hackathon, the effort concluded with the content having been ported to Github in markdown format and having an aesthetic Hugo-Docsy-based website live.
+
+### Conversion Tooling
+
+Porting DocBook XML content in Paligo to Doc-as-Code-friendly markdown format required some level of automation, and the obvious choice for the conversion was `pandoc`. It is an open-source universal document converter that supports both DocBook and markdown. However, you cannot simply run a `pandoc` command and expect the XML to be converted into a desired form of markdown. Not all DocBook tags are easily convertible to markdown. There are many attributes that XML has and markdown does not, and on top of that, Paligo proprietary identifiers require different treatment.
+
+Therefore, the focus of the Hackathon team was to refine the XML file prior to running the `pandoc` command, and post conversion, cleaning up the tags in the converted markdown files. Patching the XML file included, but was not limited to, replacing attributes with those that `pandoc` can identify. The post-conversion refinement dealt mostly with polishing how content is displayed on the website.
+
+The result was a Python script to solve this problem and make the conversion automated.
+
+### CI/CD Pipeline
+
+We chose [Hugo](https://gohugo.io/) static site generator with [Docsy](https://www.docsy.dev/) theme for building the documentation website for its rich feature set. Hugo is faster, offers a rich feature set for expansion, and has an active community for support. Docsy offers features such as dark theme, RSS feed support, and swagger integration for APIs, which we plan to introduce to the site at some point. [Netlify](https://www.netlify.com/) is faster to launch and easier to integrate with Github repository for hosting preview and production builds. The publishing flow has been designed in such a way that each Pull Request would generate a preview build so reviewers can see the content as it would appear on the website. This mechanism accelerated our documentation publishing pace without sacrificing quality.
+
```

Comment:
There's an image on the blog that's missing here; can you add it? It's https://sysdig.com/wp-content/uploads/adopting-doc-as-code-01.png